### ISSUE TITLE
Killer Instinct Sound Glitches & Freezing

### ISSUE THREAD

**Issue by MordecaiWalfish** [MiSTer-devel/SNES_MiSTer#63]
Tested with latest official SNES release (3-3-19) and more recent builds compiled by James.

Steps to reproduce: 
Start single player game and play normally. If you play long enough, the game will eventually freeze and/or the audio will stutter in a loop. Sometimes sound will glitch and gameplay will continue for a round or two until the whole game eventually freezes.

This seems to happen more commonly if you mash all the buttons while playing. Try setting turbo to all buttons and holding them down in-between rounds, etc and you will likely start a round with the sound glitching. Might be a coincidence that this happening, but it seemed more common when mashing buttons.

**Comment by James-F2** [comment]
I confirm.

Sometimes it will have audio glitches or play different samples altogether.
Sometimes, the game will freeze during a fight.

**Comment by srg320** [comment]
Yes, the problem is in SPC700, I am working on it now.

**Comment by srg320** [comment]
Try the latest release.

**Comment by MordecaiWalfish** [comment]
I tested with the latest release and the same issues occur. 
sound cut out before i even got to the character selection the first time i tried. reset and sound came back but then about 5 fights into single player mode the game froze

**Comment by James-F2** [comment]
Yes this still happens, especially on a fresh boot of the core.

**Comment by James-F2** [comment]
Seems to be fixed in 20190609 release.
I don't see SPC700 commits but I don't experience freezes or audio glitches anymore.
Maybe it was memory related outside the core?

Needs to be confirmed by at least one more person to close issue.

**Comment by Hewhoisred** [comment]
Played about an hour, completed the game.  No issues that I could see.

**Comment by srg320** [comment]
I also do not see the issue.

**Comment by James-F2** [comment]
This is very strange.
I just checked previous releases from 2019.05.10 and they all have issues with this game, but not 20190609 release.
I don't see a particular commit that might have fixed this.

**Comment by srg320** [comment]
@James-F2, this commit fafa721d55c33237c927f7690e771906987105cf

**Comment by James-F2** [comment]
Ah, yes, I thought it was DSP1-4 special chip related, and not general signal processing.
Absolutely amazing work.

**Comment by MordecaiWalfish** [comment]
I can confirm this issue is no longer occurring, neither the sound problem or freezing.

I played through for over an hour without any issues, fell asleep for 6 hours with the MiSTer running the demo loop and when I came back to it everything was still functioning properly.

Thanks for your time and attention in addressing this, and your continuously impressive work with the console cores, @srg320 . I dont know how tough it would be to address the turbografx16 issues, but there has been a long-standing graphics problem with that core, that affects some pretty big games like SF2:CE. Seeing that core come to parity with the current quality/stability of snes/nes/genesis cores would be a big step forward IMO.

I am closing this issue.
